packages/dd-trace/src/appsec/iast/vulnerability-reporter.js

```diff
@@ -17,9 +17,15 @@ let resetVulnerabilityCacheTimer
 let deduplicationEnabled = true
 
 function addVulnerability (iastContext, vulnerability) {
-  if (vulnerability && vulnerability.evidence && vulnerability.type &&
-    vulnerability.location) {
-    if (iastContext && iastContext.rootSpan) {
+  if (vulnerability?.evidence && vulnerability?.type && vulnerability?.location) {
+    if (deduplicationEnabled && isDuplicatedVulnerability(vulnerability)) return
+
+    VULNERABILITY_HASHES.set(`${vulnerability.type}${vulnerability.hash}`, true)
+
+    if (iastContext?.rootSpan) {
+      keepTrace(iastContext.rootSpan, SAMPLING_MECHANISM_APPSEC)
+      standalone.sample(iastContext.rootSpan)
```

Comment:
Could you move these two out of the `if`, and remove them from `sendVulnerabilities` method (lines 57 and 58)?
```suggestion
    keepTrace(iastContext.rootSpan, SAMPLING_MECHANISM_APPSEC)
    standalone.sample(iastContext.rootSpan)
    
    if (iastContext?.rootSpan) {

```

Reply:
No I cannot. If these calls were moved outside the conditional, there would be no certainty that a span exists - it is necessary to take into account the cases of vulnerabilities outside the request, in which there is no span and it is created on the fly in the `sendVulnerabilities` method.

Reply:
In order to avoid having multiple calls to `keepTrace` and `standalone.sample`, span on the fly creation can be moved to `addVulnerability` method.

WDYT?

Reply:
ah true!
> span on the fly creation can be moved to addVulnerability method.

i like it!!